src/ipc/ipc4/helper.c

```diff
@@ -20,8 +20,14 @@
 #include <sof/platform.h>
 #include <sof/schedule/ll_schedule_domain.h>
 #include <rtos/wait.h>
-#ifdef __ZEPHYR__
-#include <adsp_memory.h> /* for IMR_BOOT_LDR_MANIFEST_BASE */
+
+/* TODO: Remove platform-specific code, see https://github.com/thesofproject/sof/issues/7549 */
+#if defined(CONFIG_SOC_SERIES_INTEL_ACE) || defined(CONFIG_INTEL_ADSP_CAVS)
+#define RIMAGE_MANIFEST 1
+#endif
+
+#if RIMAGE_MANIFEST
```

Comment:
`#ifdef`? Also you still need to check for `__ZEPHYR__` I think - as of today IPC4 can still be done with XTOS and we aren't removing that yet

Reply:
Good catch, though in this context #if works the same (the original scheme expanded it to some defined() expressions, which I then had to revert because it turns out that's a compiler warning on some toolchain or another).

But __ZEPHYR__ is implicit on the test: RIMAGE_MANIFEST is set based on detection of Zephyr SOC kconfigs.

Reply:
At least in some cases trying to do an `#if` on a symbol, that isn't defined at all, generated an error (I think it was a hard error, not a warning) - cannot check value of something, that isn't defined.
`__ZEPHYR__` - implicitly, yes... I guess it's good enough for now.